src/arrays.js

```diff
@@ -1,49 +1,65 @@
 const getNthElement = (index, array) => {
-  // your code here
+  if (index >= array.length) 
+  return array[index - array.length];
+  return array[index]
 };
 
 const arrayToCSVString = array => {
-  // your code here
+  return array.toString()
 };
 
-const csvStringToArray = string => {
-  // your code here
-};
+function csvStringToArray(string) {
+  return string.split(",", 5)
+}
 
-const addToArray = (element, array) => {
-  // your code here
+const addToArray = (elementX, arrayX) => {
+  arrayX.push(elementX)
 };
 
-const addToArray2 = (element, array) => {
-  // your code here
+const addToArray2 = (NEWelement, ORIGarray) => {
+  let addToArray3;
+  addToArray3 = ORIGarray.concat(NEWelement)
+  return addToArray3
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+ return array.splice(index,1)
 };
 
+
 const numbersToStrings = numbers => {
-  // your code here
+  return numbers.map(String)
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  return strings.map(function(A){return A.toUpperCase();})
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+   return strings.map(function(A){return A.split("").reverse().join("");})
 };
 
 const onlyEven = numbers => {
-  // your code here
+//const numbers2 = numbers.filter(a => a % 2 === 0); works, replaced filter with map,but this gives T or F//
+return numbers.filter(a => a % 2 === 0);
 };
 
+//splice() method mutates an array and returns the removed items. Splice example with no mutation.
 const removeNthElement2 = (index, array) => {
-  // your code here
+let newArray = [...array]
+if (index >= 0) {
+    newArray.splice(index, 1)
+    return newArray
+  }
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+  let results = []
+  for (let i = 0; i <strings.length; i++) {
+    if ((strings[i].startsWith('a')) || (strings[i].startsWith('e')) || (strings[i].startsWith('i')) || (strings[i].startsWith('o')) || (strings[i].startsWith('u')))
+    results.push(strings[i])
+    }
+    return results
```

Comment:
The formatting on this is really something! Remember eslint is you friend and will sort out all this stuff for you.
That aside, this could use some work as it currently has a lot of repetition and is also only working for lower case letters since thats all you have defined. The cleanest way to solve this is to filter through the array suing the filter method, and use pattern recognition (namely regex) to find the elements that match the criteria you want. So:

```
  return strings.filter(string => string.match(/^[aeiouAEIOU]/));
```
Whats happening here is the filter function is looking at all elements in the given array that match the pattern given in the brackets. Regex is quite a big topic in and of itself so I wont go into detail here, but in essence this pattern means 'match anything that has any one of this group of letters at its start'
As this includes both the upper and lower cases, the test for case insensitivity that is currently marked as skipped can be switched back on, and it will now also pass :)

Reply:
Something has gone wrong here as I did complete all the katas, and all tests were passing with none switched off with 'xit'. I must have forgotten to add all my files to the repo!!! I will check this. I did ask the question about regex as I found this regex solution on stackoverflow.  I was told by tutors to avoid regex at this stage as we haven't covered this topic. I'll check it out. I knew having seen the regex solution, just how cleaner it was than my solution which is even longer after I added the Upper Case letters too!! : )